### PR TITLE
Added Samba in rules

### PR DESCRIPTION
**Pull request by glpatcern**
#### Pull Request (PR) description
These are the nft in rules to operate a Samba cluster, including `ctdb`. Tested at CERN.
The `ctdb` opening is optional and a parameter is provided for that.

#### This Pull Request (PR) fixes the following issues
N/A

### CODE REVIEW

File: manifests/rules/samba.pp [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      content => 'tcp dport {139,445} accept',
+  }
+
+  nftables::rule {
+    'default_in-netbios_udp':
+      content => 'udp dport {137,138} accept',
+  }
+
+  nftables::rule {
+    'default_in-ctdb':
```
Comment (nbarrientos) [inline manifests/rules/samba.pp:14]: If CTDB is not strictly necessary to run a Samba server I'd create a separate rule class for it (`nftables::rules::ctdb`) and move the responsibility of including it or not to the user of the module.

Comment (glpatcern) [reply manifests/rules/samba.pp:14]: I was rather thinking to have this parameterized with a `boolean` (working on that, that's why I switched the PR to draft).

IMHO A ctdb rule alone makes little sense as you can't run it without samba.

File: manifests/rules/samba.pp [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# @summary manage Samba, the suite to allow Windows file sharing on Linux resources.
+# @params ctdb Enable ctdb-driven clustered Samba setups.
```
Comment (nbarrientos) [inline manifests/rules/samba.pp:2]:
```suggestion
# @param ctdb Enable ctdb-driven clustered Samba setups.
```

**Review by nbarrientos — CHANGES_REQUESTED** [review]

File: manifests/rules/samba.pp [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# @summary manage Samba, the suite to allow Windows file sharing on Linux resources.
+# @param ctdb Enable ctdb-driven clustered Samba setups.
+
```
[the remark this answers is not in the document]
Comment (glpatcern) [reply manifests/rules/samba.pp]: oh gosh :-P ok, just squashed those last bits

**Review by nbarrientos — APPROVED** [review]
LGTM, thanks a lot for the patch! The `REFERENCE` can be regenerated later on. Asking for a second opinion, though.

File: manifests/rules/samba.pp [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    'default_in-netbios_tcp':
+      content => 'tcp dport {139,445} accept',
+  }
+
+  nftables::rule {
+    'default_in-netbios_udp':
+      content => 'udp dport {137,138} accept',
+  }
+
+  if ($ctdb) {
```
Comment (duritong) [inline manifests/rules/samba.pp:16]: imho there is not need for the parantheses

**Review by duritong — CHANGES_REQUESTED** [review]

File: manifests/rules/samba.pp (hunk 4 above)
Comment (glpatcern) [reply manifests/rules/samba.pp]: indeed, that was a left-over - removed now

**Pull request merged**